runtime/commonMain/src/kotlinx/serialization/SerialDescriptorBuilder.kt

```diff
@@ -229,6 +245,7 @@ private class SerialDescriptorImpl(
         if (this === other) return true
         if (other !is SerialDescriptor) return false
         if (serialName != other.serialName) return false
```

Comment:
Our resolution was the following:
1) Compare identities
2) Compare serial names
3)  Cache and compare hashcodes 
4) Compare type parameters
5) Compare element names
6) Compare element kinds

Also, please reflect these parameters in hashcode computation as well